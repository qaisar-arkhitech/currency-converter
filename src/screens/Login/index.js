import React, {useContext} from "react"
import {View, Text, TextInput, TouchableOpacity} from "react-native"
import PropTypes from "prop-types"
import {useRoute} from "@react-navigation/native"
import Color from "color"
import {KeyboardAwareScrollView} from "react-native-keyboard-aware-scroll-view"
import {HeaderBar} from "../../components/HeaderBar"
import {ThemeContext} from "../../ContextUtils/ThemeContext"
import themeStyles from "./styles"
import {Logo} from "../../components/Logo"

// Login Component content
export const Login = () => {
  const {styleableTheme} = useContext(ThemeContext)
  const styles = themeStyles(styleableTheme)
  const {name} = useRoute()

  return (
    <View style={styles.root}>
      <HeaderBar isHeaderShow={false} title={name} />
      <KeyboardAwareScrollView
        scrollEnabled
        behavior="padding"
        style={styles.container}
        contentContainerStyle={styles.container}>
        <View style={styles.content}>
          <Logo />
          <TextInput
            style={styles.input}
            placeholder="Email"
            placeholderTextColor={Color(styleableTheme.A700).darken(0.6).hex()}
            autoCapitalize="none"
            autoCompleteType="email"
            autoFocus
          />
          <TextInput
            style={styles.input}
            placeholder="Password"
            placeholderTextColor={Color(styleableTheme.A700).darken(0.6).hex()}
            autoCapitalize="none"
            secureTextEntry
          />
          <TouchableOpacity style={styles.submitButton} onPress={() => {}}>
            <Text style={styles.submitButtonText}> Submit </Text>
          </TouchableOpacity>
        </View>
      </KeyboardAwareScrollView>
    </View>
  )
}
// Login Proptypes
Login.propTypes = {}

// Login Default props
Login.defaultProps = {}

export default Login
