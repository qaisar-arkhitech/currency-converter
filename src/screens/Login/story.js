import React from "react"
import Login from "."

export default {
  title: "Login",
  storyFn() {
    return (
      <div>
        <Login />
      </div>
    )
  },
}
