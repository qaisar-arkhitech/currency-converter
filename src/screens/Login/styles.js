import Color from "color"
import {StyleSheet} from "react-native"

// Login: Styling here
const styles = (theme) => {
  return StyleSheet.create({
    root: {
      flex: 1,
      backgroundColor: theme[500],
    },
    container: {
      flex: 1,
    },
    content: {
      flex: 1,
      justifyContent: "center",
    },
    input: {
      padding: 14,
      paddingLeft: 10,
      paddingRight: 10,
      fontSize: 18,
      color: Color(theme.A700).darken(0.6).hex(),
      backgroundColor: "#FFF",
      marginVertical: 10,
      marginHorizontal: 20,
      flexDirection: "row",
      borderRadius: 5,
      borderWidth: StyleSheet.hairlineWidth,
      borderColor: theme[200],
    },
    submitButton: {
      backgroundColor: theme[900],
      padding: 10,
      margin: 15,
      alignItems: "center",
      height: 40,
      borderRadius: 20,
      overflow: "hidden",
    },
    submitButtonText: {
      color: theme[50],
      fontSize: 18,
    },
  })
}
export default styles
