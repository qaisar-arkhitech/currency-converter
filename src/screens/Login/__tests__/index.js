import React from "react"
import renderer from "react-test-renderer"
import Login from ".."

const createProps = () => ({
  classes: {},
})

describe("<Login />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<Login {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
