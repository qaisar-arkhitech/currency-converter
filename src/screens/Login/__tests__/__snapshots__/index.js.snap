// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Does not explode 1`] = `
<View>
  <Text>
    Login
  </Text>
</View>
`;
